### ISSUE TITLE
Custom Port Support

### ISSUE THREAD

**Issue by prototype464** [dotStart/Beacon#8]
Trying to play Subnautica (Nitrox mod with a friend, allows multiplayer) that runs on the port 11000 and can't because this feature was removed from Beacon and the old version of the program no longer runs.

Terraria should also be added as a preset as it runs under 7777, although, for people wanting to run multiple servers on a machine (I.E. a resource world in Terraria on a different port like 7778) is no longer possible without such an important feature. I don't understand why it was removed. Please add it back!

**Comment by dotStart** [comment]
I believe I've explained this a bit back on my Discord but since this hasn't been clearly outlined anywhere on the project pages as far as I can tell:

The custom port functionality never made it into the 2.x rewrite as I deemed it to be a feature for advanced users (which isn't exactly the target audience of the tool, to begin with). The original plan, however, was to re-introduce some form of an editor or advanced mode for people to curate their own port listings and even share them (some logic in that regard is already present within the code right now actually).

This being said: I currently have some plans to bring back broader support for applications (which may include support for custom port mappings in the future) but given that this is an open-source project and thus primarily a side project that I work on when I feel like it, this may still be away a couple of months. In the meantime, I can only recommend to directly forward ports in your router for more advanced use cases.

For tracking purposes, this ticket will remain open (and pinned). Updates will most likely be announced on my Discord as they happen (as usual).
